m2cgen/assemblers/boosting.py

```diff
@@ -4,12 +4,13 @@
 from m2cgen.assemblers import utils
 from m2cgen.assemblers.base import ModelAssembler
 
+MAX_LEAVES_PER_SUBROUTINE = 2000
```

Comment:
Would be great if you could add some considerations about this value. Maybe what are the values for estimators/depth you tested it against.

Reply:
I've tested against a bunch of parameters and haven't found a satisfying answer here, the amount of code generated seems to depend not only on the number of leaves but also the number of estimators. 

Searching of the parameter space, here are the breaking points for the given parameters:

```
| total_num_leaves | n_estimators | max_leaves |
|------------------|--------------|------------|
| 6448             | 10           | 3300       |
| 11125            | 20           | 2800       |
| 14344            | 30           | 2400       |
| 10986            | 50           | 2900       |
| 16650            | 100          | 2200       |
| 16908            | 150          | 1500       |
| 16933            | 175          | 1300       |
| 16958            | 200          | 1200       |
| 17008            | 250          | 900        |
| 17058            | 300          | 800        |
```